normalizer/normalizer.bal

```diff
@@ -0,0 +1,64 @@
+import ballerina/http;
+
+type ZendeskResponse record {
+    record {|
+        string url;
+        int id;
+        string subject;
+    |} ticket;
+};
+
+service /api/v1 on new http:Listener(8080) {
+    
+    resource function post ticket(@http:Payload json|xml request) returns xml|json|error {
+        json zendeskRequest = {};
+        if request is json && validateJsonRequest(request) {
+            zendeskRequest = {
+                ticket: {
+                    subject: check request.subject, 
+                    comment: {
+                        body: check request.comment
+                        }
+                    }
+                };
+        }
+        if request is xml && validateXmlRequest(request) {
+            zendeskRequest = {
+                ticket: {
+                    subject: (request/<subject>).data(), 
+                    comment: {
+                        body: (request/<comment>).data()
+                        }
+                    }
+                };
+        }
+        if zendeskRequest != {} {
+            http:Client zendeskClient = check new ("http://api.zendesk.com.balmock.io");
+            ZendeskResponse result = check zendeskClient->/api/v2/tickets.post(zendeskRequest);
+            if request is xml {
+                return xml `
+                    <response>
+                    <id>${result.ticket.id}</id>
+                    <subject>${result.ticket.subject}</subject>
+                    <url>${result.ticket.url}</url>
+                    </response>`;
+            } else {
+                return {
+                    id: result.ticket.id,
+                    subject: result.ticket.subject,
+                    url: result.ticket.url
+                };
```

Comment:
Do we need to have different response formats? IMO since it's not part of the Normalizer pattern it would be more clear to return a common response irrespective of the request format

Reply:
@manuranga wdyt?

Reply:
+1 for @ushirask's suggestion.